### PR TITLE
Allow env access in Flows Run Script operation

### PR DESCRIPTION
**Pull request by licitdev**
## Description

Implements #16108. Allow access to environment variables via `data['$env']` or `data.$env` or `process.env` in Flows Run Script operation.

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [x] Documentation was added/updated. PR: https://github.com/directus/docs/pull/185

### CODE REVIEW

**Review by rijkvanzanten — APPROVED** [review]

**Review by br41nslug — APPROVED** [review]
Looks like a solid approach to me 😄 I am wondering tho if we should stay on the safe side and block `KEY`/`SECRET` (or possibly database credentials) from being "allowed" 🤔

**Comment by licitdev** [comment]
The `$env` variable is consistent with other variables such as `$last`, `$accountability` and `$trigger` where it can be used within the options.

There are numerous ways to accessed the variables in the run script operation as follows. 
Should the global variable be removed?

```js
module.exports = async function (data) {
	// The following logs the same value
	console.log(process.env); // Node.js method
	console.log($env); // Global variable
	console.log({{$env}}); // Parsed options
	console.log(data.$env) // Passed data, or data['$env']
};
```

**Comment by br41nslug** [comment]
> Should the global variable be removed?

Yeah i think that one might be a bit overkill, none of the others are directly accessible as global variables right?

**Comment by licitdev** [comment]
@br41nslug That has been added that in https://github.com/directus/directus/pull/16180 🤔

**Comment by br41nslug** [comment]
Ah i did not know that 😓 never mind then

**Comment by licitdev** [comment]
@br41nslug The global variable in the run script operation is indeed overkill and hard to document. Removed! 😄

File: api/src/env.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ const allowedEnvironmentVars = [
 	'EXPORT_BATCH_SIZE',
 	// flows
 	'FLOWS_EXEC_ALLOWED_MODULES',
+	'FLOWS_EXEC_ALLOWED_ENV',
```
Comment (nickrum) [inline api/src/env.ts:180]: Now that it controls access to env vars for any operation, do we want to rename this to something like `FLOWS_ALLOWED_ENV` or `FLOWS_ENV_ALLOW_LIST`?

**Review by nickrum — DISMISSED** [review]
I'm wondering if we want to remove the exec-operation-specific handling of env vars completely to make it consistent with the other operations 🤔

File: api/src/env.ts (hunk 1 above)
Comment (licitdev) [reply api/src/env.ts]: Oh yes, definitely! Updated the changes. 👍

**Comment by br41nslug** [comment]
Have change requests been resolved to merge this one @licitdev ?

**Comment by licitdev** [comment]
@br41nslug Yes they were.

**Pull request merged**